cpp/src/parquet/bloom_filter.cc

```diff
@@ -136,6 +144,15 @@ BlockSplitBloomFilter BlockSplitBloomFilter::Deserialize(
     bloom_filter.Init(header_buf->data() + header_size, bloom_filter_size);
     return bloom_filter;
   }
+  if (bloom_filter_length && *bloom_filter_length < bloom_filter_size + header_size) {
```

Comment:
The code is getting a bit confusing with `bloom_filter_length` vs. `bloom_filter_size`. Perhaps rename the latter to `bloom_filter_data_size`?

Reply:
Also, why the inequality? We should have `*bloom_filter_length == bloom_filter_data_size + header_size`.

Reply:
They should be equal. let me check them

Reply:
Checked now